src/openrct2/object/RideObject.cpp

```diff
@@ -76,26 +77,18 @@ void RideObject::ReadLegacy(IReadObjectContext * context, IStream * stream)
     GetStringTable()->Read(context, stream, OBJ_STRING_ID_NAME);
     GetStringTable()->Read(context, stream, OBJ_STRING_ID_DESCRIPTION);
 
-    // TODO: Move to its own function when ride construction window is merged.
-    if (gConfigInterface.select_by_track_type)
-    {
-        _legacyType.enabledTrackPieces = 0xFFFFFFFFFFFFFFFF;
-    }
-    else
+    // Add boosters if the track type is eligible
+    for (sint32 i = 0; i < 3; i++)
     {
-        // When not in select by track type mode, add boosters if the track type is eligible
-        for (sint32 i = 0; i < 3; i++)
+        if (_legacyType.ride_type[i] == RIDE_TYPE_LOOPING_ROLLER_COASTER ||
+            _legacyType.ride_type[i] == RIDE_TYPE_CORKSCREW_ROLLER_COASTER ||
+            _legacyType.ride_type[i] == RIDE_TYPE_TWISTER_ROLLER_COASTER ||
+            _legacyType.ride_type[i] == RIDE_TYPE_VERTICAL_DROP_ROLLER_COASTER ||
+            _legacyType.ride_type[i] == RIDE_TYPE_GIGA_COASTER ||
+            _legacyType.ride_type[i] == RIDE_TYPE_JUNIOR_ROLLER_COASTER)
```

Comment:
Moving it to a function is good, but it would be more consistent to have in RideObject.cpp, since that's also where PerformRCT1CompatibilityFixes resides. (I also think it's a better location in general.)